lib/set_builder/modifier/base.rb

```diff
@@ -79,12 +81,20 @@ def errors_with_number_value(string)
 
 
       def to_s(negative=false)
-        words = negative ? [self.class.negate(operator).to_s.gsub(/_/, " ")] : [operator.to_s.gsub(/_/, " ")]
-        arguments = self.class.operators[operator] || []
-        (0...arguments.length).each do |i|
-          words << ValueMap.to_s(arguments[i], values[i])
+        arguments = self.class.parsed_operators[operator] || Arguments.new("")
+        operator.to_s.gsub(/_/, " ") + " " + arguments.to_s(values)
+      end
+
+
+
+      def self.parsed_operators
+        @parsed_operators ||= begin
+          Hash[
+            operators.map do |operator, arguments|
+              [operator, Arguments.new(arguments)]
+            end
+          ]
         end
```

Comment:
:art: You don't need the `begin....end` b/c what follows (`Hash[...]`) is a single expression, so you could just remove those.

The `]` on its own line feels strange to me — _and_ the `do...end` despite the `.map` being a succinct block concerned with returning a value (not executing an instruction with side effects).

What about:

```ruby
# lisp- and java-style line-wrapping
@parsed_operators ||= Hash[operators.map { |operator, arguments|
  [operator, Arguments.new(arguments)] }]
```

or even (requiring Ruby 2.1+):

```ruby
# jquery-style chaining
@parsed_operators ||= operators
  .map { |operator, arguments| [operator, Arguments.new(arguments)] }
  .to_h
```

I think we use both those styles more frequently than:

```ruby
@parsed_operators ||= begin
  Hash[
    # ...
  ]
end
```